lightning/src/onion_message/messenger.rs

```diff
@@ -713,6 +715,47 @@ where
 	pub fn new(
 		entropy_source: ES, node_signer: NS, logger: L, message_router: MR, offers_handler: OMH,
 		custom_handler: CMH
+	) -> Self {
+		Self::new_inner(
+			entropy_source, node_signer, logger, message_router, offers_handler,
+			custom_handler, false
+		)
+	}
+
+	/// Similar to [`Self::new`], but rather than dropping onion messages that are
+	/// intended to be forwarded to offline peers, we will intercept them for
+	/// later forwarding.
+	///
+	/// Interception flow:
+	/// 1. If an onion message for an offline peer is received, `OnionMessenger` will
+	///    generate an [`Event::OnionMessageForOfflinePeer`]. Event handlers can
+	///    then choose to persist this onion message for later forwarding, or drop
+	///    it.
+	/// 2. When the offline peer later comes back online, `OnionMessenger` will
+	///    generate an [`Event::OnionMessagePeerConnected`]. Event handlers will
+	///    then fetch all previously intercepted onion messages for this peer.
+	/// 3. Once the stored onion messages are fetched, they can finally be
+	///    forwarded to the now-online peer via [`Self::forward_onion_message`].
+	///
+	/// # Note
+	///
+	/// LDK will not rate limit how many [`Event::OnionMessageForOfflinePeer`]s
+	/// are generated, so it is the caller's responsibility to limit how many
```

Comment:
Technically, that's correct, but we do have a hard cap on the total number of events we can store (enforced by `enqueue_events`). Do you think it would be a good idea to update the docs to clarify that users can't set an infinitely high cap on the number of events they can receive?

Reply:
The events queue should never get close to filling up, if it is we have bigger problems and the user's offer implementation is probably broken :) Not sure what you mean about users setting a cap on the number of events they can receive.

Reply:
Well, that makes sense! The upper cap by `enqueue_events` is more like a fail-safe, than a hard cap, so it makes sense to not mention it here!

> Not sure what you mean about users setting a cap on the number of events they can receive.

So I was trying to point to this line
> "it is the caller's responsibility to limit how many..."

But considering the context, I don't think we need to make any changes here.
Thanks for clarifying it! ✌️